src/EFCore/ChangeTracking/Internal/NavigationFixer.cs

```diff
@@ -234,20 +242,39 @@ public virtual void NavigationReferenceChanged(InternalEntityEntry entry, INavig
                     {
                         _inFixup = true;
 
-                        // Null FKs and navigations of dependents that have been removed, unless they
-                        // have already been changed.
-                        ConditionallyNullForeignKeyProperties(oldTargetEntry, entry, foreignKey);
+                        if (navigationBase is ISkipNavigation skipNavigation)
+                        {
+                            var associationEntry = FindAssociationEntry(entry, oldTargetEntry, skipNavigation);
+                            associationEntry?.SetEntityState(EntityState.Deleted);
 
-                        if (inverse != null
-                            && ReferenceEquals(oldTargetEntry[inverse], entry.Entity)
-                            && (!foreignKey.IsOwnership
-                                || (oldTargetEntry.EntityState != EntityState.Deleted
-                                    && oldTargetEntry.EntityState != EntityState.Detached)))
+                            if (skipNavigation.Inverse.IsCollection)
```

Comment:
```suggestion
                            if (skipNavigation.Inverse?.IsCollection)
```

Although it's weird, I had some tests at one point that managed to set up skipNav's with strange inverses including `null`. So suggest the `?` form. Similarly in a few places below. That said, in this particular case you may be able to ignore the inverse and carry on rather than throwing in the `else`?

Reply:
See discussion from last week in team chat. @AndriySvyryd asserted that the Inverse will not be null. It may be that the cases you were hitting were while the model was being built.

Reply:
Inverse will always be non-null after model validation